python_modules/dagster/dagster/core/instance/__init__.py

```diff
@@ -130,24 +130,42 @@ def __init__(self, instance):
 
     def emit(self, record):
         from dagster.core.events.log import construct_event_record, StructuredLoggerMessage
+        from dagster.core.events import EngineEventData
 
-        try:
-            event = construct_event_record(
-                StructuredLoggerMessage(
-                    name=record.name,
-                    message=record.msg,
-                    level=record.levelno,
-                    meta=record.dagster_meta,
-                    record=record,
-                )
+        event = construct_event_record(
+            StructuredLoggerMessage(
+                name=record.name,
+                message=record.msg,
+                level=record.levelno,
+                meta=record.dagster_meta,
+                record=record,
             )
+        )
 
+        try:
             self._instance.handle_new_event(event)
-
-        except Exception as e:  # pylint: disable=W0703
-            logging.critical("Error during instance event listen")
-            logging.exception(str(e))
-            raise
+        except Exception as e:
+            if "root" in self._instance.managed_python_loggers:
```

Comment:
what exactly does `"root"` imply here? Feels like this is making an assumption that could change over time